ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/WatchedImages/WatchedImagesModal.tsx

```diff
@@ -8,19 +15,53 @@ export type WatchedImagesModalProps = {
 };
 
 function WatchedImagesModal({ defaultWatchedImageName, isOpen, onClose }: WatchedImagesModalProps) {
+    const { data, mutate, isSuccess, isLoading, isError, error, reset } = useRestMutation(
+        (name: string) => watchImage(name)
+    );
+
+    function onModalClose() {
+        onClose();
+        reset();
+    }
+
     return (
         <Modal
             title="Manage watched images"
             isOpen={isOpen}
-            onClose={onClose}
+            onClose={onModalClose}
             variant="medium"
+            showClose={!isLoading}
+            onEscapePress={isLoading ? noop : onModalClose}
             actions={[
-                <Button key="Close" onClick={onClose}>
+                <Button key="Close" onClick={onModalClose} isDisabled={isLoading}>
                     Close
                 </Button>,
             ]}
         >
-            <div>Image name: {defaultWatchedImageName}</div>
+            {isSuccess && (
+                <Alert
+                    variant="success"
+                    isInline
+                    title="The image was successfully added to the watch list"
+                    className="pf-u-mb-md"
```

Comment:
A comment, not necessarily a change request: recently, I tried `<Flex direction={{ default: 'column' }}>` as an alternative to PatternFly utility classes for spacing.

Reply:
Oh good call, I've been gravitating towards this as well as it tends to give more consistent results with less code.